bot/converters.py

```diff
@@ -267,37 +268,52 @@ async def convert(self, ctx: Context, arg: str) -> int:
         return snowflake
 
 
-class SourceConverter(Converter):
-    """Convert an argument into a help command, tag, command, or cog."""
+class SourceTrasnformer(app_commands.Transformer):
+    """Transform an argument into a tag, command or cog."""
 
-    @staticmethod
-    async def convert(ctx: Context, argument: str) -> SourceType:
-        """Convert argument into source object."""
-        if argument.lower() == "help":
-            return ctx.bot.help_command
+    async def transform(self, interaction: discord.Interaction, source_item: str) -> SourceType:
+        """Transform source_item into source object."""
+        if source_item.lower() == "help":
+            return interaction.client.help_command
 
-        cog = ctx.bot.get_cog(argument)
+        cog = interaction.client.get_cog(source_item)
         if cog:
             return cog
 
-        cmd = ctx.bot.get_command(argument)
+        cmd = interaction.client.get_command(source_item)
         if cmd:
             return cmd
 
-        tags_cog = ctx.bot.get_cog("Tags")
+        tags_cog = interaction.client.get_cog("Tags")
         show_tag = True
 
         if not tags_cog:
             show_tag = False
         else:
-            identifier = TagIdentifier.from_string(argument.lower())
+            identifier = TagIdentifier.from_string(source_item.lower())
             if identifier in tags_cog.tags:
                 return identifier
-        escaped_arg = escape_markdown(argument)
+        escaped_arg = escape_markdown(source_item)
 
-        raise BadArgument(
+        log.debug(
             f"Unable to convert '{escaped_arg}' to valid command{', tag,' if show_tag else ''} or Cog."
         )
+        return escaped_arg
+
+    async def autocomplete(self, interaction: discord.Interaction, value: str) -> list[app_commands.Choice[str]]:
+        """Autocompleter for `/source` command."""
+        all_cogs = interaction.client.cogs
+        tags_cog = all_cogs.get("Tags", None)
+        names = [*all_cogs.keys(), *interaction.client.all_commands.keys()]
+        if tags_cog:
```

Comment:
This currently doesn't suggest the `tag` command itself when I do `/source tag`
Instead, it only returns the cog